lib/dgram.js

```diff
@@ -543,7 +563,7 @@ Socket.prototype.addMembership = function(multicastAddress,
   this._healthCheck();
 
   if (!multicastAddress) {
-    throw new Error('multicast address must be specified');
+    throw new errors.Error('ERR_MISSING_ARGS', 'multicastAaddress');
```

Comment:
> This should be a `TypeError`.

My immediate reaction was to disagree on the grounds that a`TypeError` means an argument is the wrong type, not missing.

But `TypeError` is what V8 uses for similar situations, so I'm inclined to agree that Node.js should do the same.

```console
> [0,1,2].forEach()
TypeError: undefined is not a function
 ...
>
```